cwms-data-api/src/main/java/cwms/cda/data/dao/timeseriesprofile/TimeSeriesProfileDao.java

```diff
@@ -1,48 +1,111 @@
-					parameterString, null, null, "F","T",timeSeriesProfile.getOfficeId());
-		});
-	}
-	public void retrieveTimeSeriesProfile()
-	{
-		// TODO
-	}
-	public void deleteTimeSeriesProfile()
-	{
-		// TODO
-	}
-	public void copyTimeSeriesProfile()
-	{
-		// TODO
-	}
-	public void retrieveTimeSeriesProfiles()
-	{
-		// TODO
-	}
+public class TimeSeriesProfileDao extends JooqDao<TimeSeriesProfile> {
+    public TimeSeriesProfileDao(DSLContext dsl) {
+        super(dsl);
+    }
+
+    public void storeTimeSeriesProfile(TimeSeriesProfile timeSeriesProfile, boolean failIfExists) {
+
+        connection(dsl, conn -> {
+            List<String> parameterList = timeSeriesProfile.getParameterList();
+            StringBuilder parameterString = new StringBuilder(parameterList.get(0));
+
+            for (int i = 1; i < parameterList.size(); i++) {
+                parameterString.append(",").append(parameterList.get(i));
+            }
+            String referenceTsId = null;
+            if (timeSeriesProfile.getReferenceTsId() != null) {
+                referenceTsId = timeSeriesProfile.getReferenceTsId().getName();
+            }
+            CWMS_TS_PROFILE_PACKAGE.call_STORE_TS_PROFILE(DSL.using(conn).configuration(), timeSeriesProfile.getLocationId().getName(),
+                    timeSeriesProfile.getKeyParameter(),
+                    parameterString.toString(),
+                    timeSeriesProfile.getDescription(), referenceTsId, failIfExists?"T":"F", "T", timeSeriesProfile.getLocationId().getOfficeId());
+        });
+    }
+
+    public TimeSeriesProfile retrieveTimeSeriesProfile(String locationId, String parameterId, String officeId) {
+        return connectionResult(dsl, conn -> {
+            TS_PROFILE_T timeSeriesProfile = CWMS_TS_PROFILE_PACKAGE.call_RETRIEVE_TS_PROFILE(
+                    DSL.using(conn).configuration(), locationId, parameterId, officeId);
+            return map(timeSeriesProfile, locationId, parameterId, officeId);
+        });
+    }
+
+    public void deleteTimeSeriesProfile(String locationId, String keyParameter, String officeId) {
+        connection(dsl, conn ->
+                CWMS_TS_PROFILE_PACKAGE.call_DELETE_TS_PROFILE(DSL.using(conn).configuration(), locationId, keyParameter, "DELETE ALL",
+                        officeId));
+    }
+
+    public void copyTimeSeriesProfile(String locationId, String keyParameter, String destinationLocation, String destRefTsId, String officeId) {
+        connection(dsl, conn ->
+                CWMS_TS_PROFILE_PACKAGE.call_COPY_TS_PROFILE(DSL.using(conn).configuration(), locationId, keyParameter, destinationLocation,
+                        destRefTsId, "F", "F",
+                        officeId));
+    }
+
+    public List<TimeSeriesProfile> retrieveTimeSeriesProfiles(String locationIdMask, String parameterIdMask, String officeIdMask) {
+        return connectionResult(dsl, conn -> {
+            List<TimeSeriesProfile> timeSeriesProfileList = new ArrayList<>();
```

Comment:
Are we sure this isn't going to end up large in the future?
If so we can wait on pagination but it's generally something we want to consider.